pkg/platform/runtime/buildscript/buildscript.go

```diff
@@ -79,6 +80,10 @@ func NewScript(data []byte) (*Script, error) {
 
 	script, err := parser.ParseBytes(constants.BuildScriptFileName, data)
 	if err != nil {
+		parseErrors := errs.Unpack(err)
```

Comment:
This should use errors.As or something more determinate. The code as written assumes that the returned error is specified in a very specific way that is not enforced through the type system.